lib/swagger-oauth.js

```diff
@@ -165,6 +197,28 @@ function handleLogin() {
   return;
 }
 
+function handlePasswordFlow(auth) {
+  var tokenUri = auth.tokenUrl;
+
+  var authParams = {
+    grant_type: 'password',
+    client_id: encodeURIComponent(clientId),
```

Comment:
unfortunately cannot confirm that the basic auth header part would be working for me, was trying with your last push, but had to roll back to the . I am using the shaffer/oauth2-server-php by the way. Grant type is UserCredentials (password).